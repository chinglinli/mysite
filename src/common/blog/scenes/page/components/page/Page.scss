.page {
  float: left;
  width: 800px;

  @media (max-width: 1200px) {
    width: 100%;
  }

  article {
    padding: 2em;
    border-bottom: 1px solid #eee;
    background: white;

    header {
      h1 {
        margin: 3px 0;

        a {
          font-size: 2rem;
        }
      }
    }
  }
}
