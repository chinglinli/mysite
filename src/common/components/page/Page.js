import * as React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { Link } from 'react-router'

import ArticleContent from '../article-content'

import styles from './Page.scss'

export default class Page extends React.Component {
  static PropTypes = {
    page: ImmutablePropTypes.contains({
      app: React.PropTypes.string.isRequired,
      slug: React.PropTypes.string.isRequired,
      title: React.PropTypes.string.isRequired,
      content: React.PropTypes.content
    })
  }

  render () {
    const {page} = this.props
    if (!page) {
      return <article>讀取中……</article>
    }

    return (
      <article className={styles.root}>
        <div>
          <header className={styles.header}>
            <Link className={styles.link} to={`/${page.get('app')}/${page.get('slug')}/`}>{page.get('title')}</Link>
          </header>
        </div>
        <ArticleContent content={page.get('content')}/>
      </article>
    )
  }
}
