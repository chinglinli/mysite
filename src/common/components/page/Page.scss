@import '../../css/base';

.root {
  padding: 2em;
  border-bottom: 1px solid #eee;
  background: white;
  min-height: 50em;

  .header {
    font-size: 2em;
    font-weight: bold;
    margin-bottom: 1em;
    color: $dominant-color;

    .link {
      @extend .link;
    }
  }
}
