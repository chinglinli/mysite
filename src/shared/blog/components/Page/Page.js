import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import Link from '../../../generic/components/Link/index'

import Loading from '../Loading/index'
import RstContent from '../RstContent/index'

import styles from './Page.scss'

export default class Page extends React.PureComponent {
  static PropTypes = {
    page: ImmutablePropTypes.contains({
      app: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.content
    })
  }

  render () {
    const {page} = this.props
    if (!page) {
      return (
        <section className={styles.page}>
          <Loading/>
        </section>
      )
    }

    return (
      <section className={styles.page}>
        <article>
          <div>
            <header>
              <h1><Link to={`/${page.get('app')}/${page.get('slug')}/`}>{page.get('title')}</Link></h1>
            </header>
          </div>
          <RstContent content={page.get('content')}/>
        </article>
      </section>
    )
  }
}
