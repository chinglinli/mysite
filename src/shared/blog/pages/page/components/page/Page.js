import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import Link from '../../../../../generic/components/Link'

import Loading from '../../../../components/Loading'
import ArticleContent from '../../../../components/ArticleContent'

import styles from './Page.scss'

export default class Page extends React.PureComponent {
  static PropTypes = {
    page: ImmutablePropTypes.contains({
      app: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.content
    })
  }

  render () {
    const {page} = this.props
    if (!page) {
      return (
        <section className={styles.page}>
          <Loading/>
        </section>
      )
    }

    return (
      <section className={styles.page}>
        <article>
          <div>
            <header>
              <h1><Link to={`/${page.get('app')}/${page.get('slug')}/`}>{page.get('title')}</Link></h1>
            </header>
          </div>
          <ArticleContent content={page.get('content')}/>
        </article>
      </section>
    )
  }
}
