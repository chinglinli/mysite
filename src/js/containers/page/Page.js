import * as React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import {connect} from 'react-redux'
import {Link} from 'react-router'

import * as pageActions from '../../ducks/page/actions'

import styles from './Page.scss'


export class Page extends React.Component {
  static PropTypes = {
    page: ImmutablePropTypes.contains({
      app: React.PropTypes.string.isRequired,
      slug: React.PropTypes.string.isRequired,
      title: React.PropTypes.string.isRequired,
      content: React.PropTypes.content
    })
  }

  componentWillMount() {
    if (!this.props.page) {
      this.props.fetchPages()
    }
  }

  render() {
    const {page} = this.props
    if (!page) {
      return <article>讀取中……</article>
    }

    return (
      <article className={styles.root}>
        <div>
          <header className={styles.header}>
            <Link className={styles.link} to={`/${page.get('app')}/${page.get('slug')}/`}>{page.get('title')}</Link>
          </header>
        </div>
        <div dangerouslySetInnerHTML={{__html: page.get('content')}}/>
      </article>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    page: state.getIn(['page', 'items']).find(page => (
      page.get('app') === ownProps.params.app &&
      page.get('slug') === ownProps.params.slug
    ))
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchPages: () => dispatch(pageActions.fetchPages())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Page)
