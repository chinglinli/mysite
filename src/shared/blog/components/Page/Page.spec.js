import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Page from './Page'

test('It should render Page component with empty articles correctly', () => {
  const page = null

  const component = renderer.create(
    <Page page={page}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('It should render Page component correctly', () => {
  const page = Immutable.fromJS({
    app: 'app',
    slug: 'slug',
    title: 'title',
    content: '<div>今天天氣不錯</div>'
  })

  const component = renderer.create(
    <Page page={page}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
