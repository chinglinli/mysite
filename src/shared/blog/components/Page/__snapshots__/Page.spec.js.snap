// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Page component correctly 1`] = `
<section
  className="page"
>
  <article>
    <div>
      <header>
        <h1>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            title
          </a>
        </h1>
      </header>
    </div>
    <div
      className="content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div>今天天氣不錯</div>",
        }
      }
    />
  </article>
</section>
`;

exports[`It should render Page component with empty articles correctly 1`] = `
<section
  className="page"
>
  <div
    className="Loading__Base-s1d02ruw-0 idafHP"
  >
    <div
      className="Loading__LoadingImage-s1d02ruw-1 cjrynz"
    />
    <div
      className="Loading__Message-s1d02ruw-2 fgoCqC"
    >
      正在努力讀取中…
    </div>
  </div>
</section>
`;
